src/bpftrace.cpp

```diff
@@ -786,7 +787,12 @@ void BPFtrace::poll_perf_events(int epollfd, bool drain)
   auto events = std::vector<struct epoll_event>(online_cpus_);
   while (true)
   {
-    int ready = epoll_wait(epollfd, events.data(), online_cpus_, 100);
+    int ready;
+    do {
+      ready = epoll_wait(epollfd, events.data(), online_cpus_, 100);
+    } while (ready < 0 && errno == EINTR && !bt_sigint_called);
```

Comment:
I think the code would be easier to reason about with just a single loop around `epoll_wait` - could this change to an `if (...) continue;` instead of a `do {} while(...)`?